docs/Guides/reading-mcap-files/json-2.md

```diff
@@ -0,0 +1,184 @@
+
+```python
+pointcloud["pose"] = {
+  "position": {"x": 0, "y": 0, "z": 0},
+  "orientation": {"x": 0, "y": 0, "z": 0, "w": 1},
+}
+pointcloud["frame_id"] = "lidar"
+```
+
+We’ll leave the timestamp field for later, when we write the messages into the MCAP file.
+
+### Creating a writer
+
+We’ll start with some imports from the Python MCAP library:
+
+```python
+from mcap.mcap0.writer import Writer
+from mcap.mcap0.well_known import SchemaEncoding, MessageEncoding
+```
+
+Open a file where we’ll output our MCAP data and write our header:
+
+```python
+with open(args.output, "wb") as f:
+  writer = Writer(f)
+  writer.start("x-jsonschema", library="my-excellent-library") # can use any name, as we are not using an MCAP profile: https://www.google.com/url?q=https://github.com/foxglove/mcap/blob/main/docs/specification/appendix.md%23well-known-profiles&sa=D&source=docs&ust=1658261499762935&usg=AOvVaw2GgwRzJbSlqWOErSFru7L_
+```
+
+### Register channel
+
+Let's create a channel of messages to contain our point cloud. The schema's name and content tell Foxglove Studio that it can parse and display this message as a point cloud.
+
+```python
+with open(Path(__file__).parent / "PointCloud.json", "rb") as f:
+  schema = f.read()
+schema_id = writer.register_schema(
+  name="foxglove.PointCloud",
+  encoding=SchemaEncoding.JSONSchema,
+  data=schema,
+)
+channel_id = writer.register_channel(
+  topic="pointcloud",
+  message_encoding=MessageEncoding.JSON,
+  schema_id=schema_id,
+)
+```
+
+### Write messages
+
+Let's write a single `foxglove.PointCloud` message on the channel we just created:
+
+```python
+pointcloud["timestamp"] = {
+    "sec": int(base_timestamp.timestamp()),
+    "nsec": base_timestamp.microsecond * 1000,
+}
+
+writer.add_message(
+    channel_id,
+    log_time=int(base_timestamp.timestamp() * 1e9),
+    data=json.dumps(pointcloud).encode("utf-8"),
+    publish_time=int(base_timestamp.timestamp() * 1e9),
+)
+```
+
+Close the MCAP writer to include the summary and footer in your output MCAP file:
+
+```python
+writer.finish()
+```
+
+That’s it! We now have a valid MCAP file with a single point cloud message.
+
+### Inspect your file
+
+To inspect your MCAP file, install the [MCAP CLI tool](https://github.com/foxglove/mcap/tree/main/go/cli/mcap):
+
+```bash
+$ brew install mcap
+```
+
+Run the following commands to summarize your file’s contents and to verify that it has no issues:
+
+```bash
+$ mcap info output.mcap
+  library: my-excellent-library
+  profile: x-jsonschema
+  messages: 1
+  duration: 0s
+  start: 2011-11-04T01:36:05.987339008+11:00 (1320330965.987339008)
+  end: 2011-11-04T01:36:05.987339008+11:00 (1320330965.987339008)
+  compression:
+    zstd: [1/1 chunks] (48.09%)
+  channels:
+    (1) pointcloud  1 msgs (+Inf Hz)   : foxglove.PointCloud [jsonschema]
```

Comment:
+Inf Hz looks a little strange without explanation. The reason is because the duration of a one-message file is zero, so this is a divide by zero. We could get around this by recording two messages if it seems odd.